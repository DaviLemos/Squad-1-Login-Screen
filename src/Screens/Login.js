import React from 'react';
import { Button, Text } from '../Components/Buttons/ButtonContinue';
import { Form } from '../Components/Forms/Forms';
import {
  InputPass,
  InputText,
  DivContainerText,
  DivContainerPass,
} from '../Components/Inputs/InputBase';
import Container from '../Components/Container/Container';
import Background from '../Components/Background/Background';
import Saudacao from '../Components/Saudacao/Saudacao';
import Frase from '../Components/Frase/Frase';
import Imagem from '../Components/Imagem/Imagem';
import Logo from '../Components/Logo/Logo';
import ContentContainer from '../Components/ContentContainer/ContentContainer';
import image from '../Images/image2.jpg';
import LogoCompassoBranco from '../Images/Logo-Compasso-Branco.svg';
import { User, Lock } from 'react-feather';
import { Titulo } from '../Components/Titulo/Titulo';
import { useFormik } from 'formik';
import validate from '../helper/validate.helper';

function Login() {
  const [usuario, setUsuario] = React.useState('');
  const [senha, setSenha] = React.useState('');

  const formik = useFormik({
    initialValues: {
      usuario: '',
      senha: '',
    },
    validate,
    onSubmit: (values) => {
      setUsuario(values.usuario);
      setSenha(values.senha);
    },
  });

  return (
    <Container>
      <Background>
        <ContentContainer>
          <Saudacao>Ol치,</Saudacao>
          <Frase>
            Para continuar navegando de forma segura, efetue o login na rede.
          </Frase>

          <Form onSubmit={formik.handleSubmit}>
            <Titulo>Login</Titulo>

            <DivContainerText>
              <InputText
                placeholder="Usu치rio"
                id="usuario"
                name="usuario"
                required
                value={formik.values.usuario}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />

              <User
                color="#E0E0E0"
                size={24}
                style={{ marginTop: 'auto', marginBottom: 'auto' }}
              />
            </DivContainerText>

            <DivContainerPass>
              <InputPass
                placeholder="Senha"
                id="senha"
                name="senha"
                required
                value={formik.values.senha}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />

              <Lock
                color="#E0E0E0"
                size={24}
                style={{ marginTop: 'auto', marginBottom: 'auto' }}
              />
            </DivContainerPass>
            <div
              style={{
                color: '#E9B425',
                textAlign: 'center',
              }}
            >
              {formik.touched.usuario &&
                formik.errors.usuario &&
                'Ops, usu치rio inv치lido. Tente novamente.'}
              {formik.touched.senha &&
                formik.errors.senha &&
                formik.errors.senha}
            </div>
            <Button>
              <Text>Continuar</Text>
            </Button>
          </Form>
        </ContentContainer>
      </Background>
      <Imagem url={image} className="img_notebook" alt="Notebook">
        <Logo
          src={LogoCompassoBranco}
          className="img_Logo"
          alt="Logo Compasso Branco"
        />
      </Imagem>
    </Container>
  );
}

export default Login;
