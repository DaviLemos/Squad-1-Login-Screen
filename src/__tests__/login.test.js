// * React * //
import React from 'react';
// * Test * //
import {
  render,
  screen,
  fireEvent,
  userEvent,
  waitFor,
  act,
} from '@testing-library/react';
import user from '@testing-library/user-event';
// * Component * //
import Login from '../Screens/Login';

jest.mock('react-router-dom', () => {
  // Require the original module to not be mocked...
  const originalModule = jest.requireActual('react-router-dom');

  return {
    __esModule: true,
    ...originalModule,
    // add your noops here
    useNavigate: jest.fn(() => 'bar'),
  };
});

describe('Login Component:', () => {
  it('Should render correctly', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });
  describe('Solutation Component:', () => {
    it('Have Solutation', () => {
      render(<Login />);
      expect(screen.getByTestId('solutation')).toBeInTheDocument();
    });
  });

  describe('Title Component:', () => {
    it('Have Title Login', () => {
      render(<Login />);

      expect(screen.getByText('Login')).toBeInTheDocument();
    });

    it('Title Have Style', () => {
      render(<Login />);

      expect(screen.getByTestId('login')).toHaveStyle({
        color: '#E0E0E0',
        fontFamily: 'Mark Pro',
        fontSize: '30px',
      });
    });
  });

  describe('Inputs Component:', () => {
    it('Have Two Inputs', () => {
      render(<Login />);

      expect(screen.getByTestId('input-user')).toBeInTheDocument();
      expect(screen.getByTestId('input-password')).toBeInTheDocument();
    });

    it('Have Two Icons', () => {
      render(<Login />);

      expect(screen.getByTestId('icon-user')).toBeInTheDocument();
      expect(screen.getByTestId('icon-password')).toBeInTheDocument();
    });

    it('Input Have Style', () => {
      render(<Login />);

      expect(screen.getByTestId('input-user')).toHaveStyle({
        width: '310px',
        borderRadius: '50px',
        color: '#E0E0E0',
      });
      expect(screen.getByTestId('input-password')).toHaveStyle({
        width: '310px',
        borderRadius: '50px',
        color: '#E0E0E0',
      });
    });

    it('Updates onChange', async () => {
      const promise = Promise.resolve();
      const { getByTestId } = render(<Login />);

      const userInput = getByTestId('input-user');
      const passwordInput = getByTestId('input-password');

      user.type(userInput, 'davi@furg.br');
      user.type(passwordInput, '1@5aeD38');

      expect(userInput.value).toBe('davi@furg.br');
      expect(passwordInput.value).toBe('1@5aeD38');
      await act(() => promise);
    });
  });

  describe('Button Component:', () => {
    it('Have Button', () => {
      render(<Login />);

      expect(screen.getByTestId('button-login')).toBeInTheDocument();
    });

    it('Have Text "Continuar" in button', () => {
      render(<Login />);

      expect(screen.getByText('Continuar')).toBeInTheDocument();
    });
    it('Button Have Style', () => {
      render(<Login />);

      expect(screen.getByTestId('button-login')).toHaveStyle({
        background: 'linear-gradient(90deg, #FF2D04 0%, #C13216 100%)',
        width: '379px',
        borderRadius: '50px',
        height: '67px',
        cursor: 'pointer',
        border: 'none',
      });
    });
  });
});

