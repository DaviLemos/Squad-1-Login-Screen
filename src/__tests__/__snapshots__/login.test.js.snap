// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component: Should render correctly 1`] = `
<div>
  <div
    class="sc-iCfMLu kuvhHl"
  >
    <div
      class="sc-furwcr gqtzCm"
    >
      <div
        class="sc-crHmcD juGKFM"
      >
        <h1
          class="sc-pVTFL jPPtoB"
          data-testid="solutation"
        >
          Ol치,
        </h1>
        <p
          class="sc-jrQzAO enGijf"
        >
          Para continuar navegando de forma segura, efetue o login na rede.
        </p>
        <form
          class="sc-dkPtRN fLQwHi"
        >
          <span
            class="sc-bqiRlB ghImxJ"
            data-testid="login"
          >
            Login
          </span>
          <div
            class="sc-jRQBWg clacSb"
          >
            <input
              class="sc-hKwDye fzXzce"
              data-testid="input-user"
              id="usuario"
              name="usuario"
              placeholder="Usu치rio"
              required=""
              type="email"
              value=""
            />
            <svg
              data-testid="icon-user"
              fill="none"
              height="24"
              stroke="#E0E0E0"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-top: auto; margin-bottom: auto;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
          </div>
          <div
            class="sc-gKclnd cArwuP"
          >
            <input
              class="sc-eCImPb kUijYi"
              data-testid="input-password"
              id="senha"
              name="senha"
              placeholder="Senha"
              required=""
              type="password"
              value=""
            />
            <svg
              data-testid="icon-password"
              fill="none"
              height="24"
              stroke="#E0E0E0"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-top: auto; margin-bottom: auto;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="11"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="11"
              />
              <path
                d="M7 11V7a5 5 0 0 1 10 0v4"
              />
            </svg>
          </div>
          <span
            class="sc-egiyK hHXGfM"
          />
          <button
            class="sc-bdvvtL kFtYDH"
            data-testid="button-login"
            type="submit"
          >
            <p
              class="sc-gsDKAQ gMaWpp"
            >
              Continuar
            </p>
          </button>
        </form>
      </div>
    </div>
    <div
      alt="Notebook"
      class="sc-kDTinF bIyCkS img_notebook"
    >
      <img
        alt="Logo Compasso Branco"
        class="sc-iqseJM krDwjj img_Logo"
        src="Logo-Compasso-Branco.svg"
      />
    </div>
  </div>
</div>
`;

exports[`Login Component: Solutation Component: Have Solutation  1`] = `undefined`;

exports[`Login Component: Solutation Component: Should render correctly 1`] = `
<div>
  <div
    class="sc-iCfMLu kuvhHl"
  >
    <div
      class="sc-furwcr gqtzCm"
    >
      <div
        class="sc-crHmcD juGKFM"
      >
        <h1
          class="sc-pVTFL jPPtoB"
        >
          Ol치,
        </h1>
        <p
          class="sc-jrQzAO enGijf"
        >
          Para continuar navegando de forma segura, efetue o login na rede.
        </p>
        <form
          class="sc-dkPtRN fLQwHi"
        >
          <span
            class="sc-bqiRlB ghImxJ"
            data-testid="login"
          >
            Login
          </span>
          <div
            class="sc-jRQBWg clacSb"
          >
            <input
              class="sc-hKwDye fzXzce"
              data-testid="input-user"
              id="usuario"
              name="usuario"
              placeholder="Usu치rio"
              required=""
              type="email"
              value=""
            />
            <svg
              data-testid="icon-user"
              fill="none"
              height="24"
              stroke="#E0E0E0"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-top: auto; margin-bottom: auto;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
          </div>
          <div
            class="sc-gKclnd cArwuP"
          >
            <input
              class="sc-eCImPb kUijYi"
              data-testid="input-password"
              id="senha"
              name="senha"
              placeholder="Senha"
              required=""
              type="password"
              value=""
            />
            <svg
              data-testid="icon-password"
              fill="none"
              height="24"
              stroke="#E0E0E0"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-top: auto; margin-bottom: auto;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="11"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="11"
              />
              <path
                d="M7 11V7a5 5 0 0 1 10 0v4"
              />
            </svg>
          </div>
          <span
            class="sc-egiyK hHXGfM"
          />
          <button
            class="sc-bdvvtL kFtYDH"
            data-testid="button-login"
            type="submit"
          >
            <p
              class="sc-gsDKAQ gMaWpp"
            >
              Continuar
            </p>
          </button>
        </form>
      </div>
    </div>
    <div
      alt="Notebook"
      class="sc-kDTinF bIyCkS img_notebook"
    >
      <img
        alt="Logo Compasso Branco"
        class="sc-iqseJM krDwjj img_Logo"
        src="Logo-Compasso-Branco.svg"
      />
    </div>
  </div>
</div>
`;
